maitake/src/wait/cell.rs

```diff
@@ -7,14 +7,35 @@ use crate::{
         },
     },
     util::tracing,
-    wait::{self, WaitResult},
 };
 use core::{
+    future::Future,
     ops,
-    task::{Poll, Waker},
+    pin::Pin,
+    task::{Context, Poll, Waker},
 };
 use mycelium_util::{fmt, sync::CachePadded};
 
+/// An error indicating that a [`WaitCell`] or [`WaitQueue`] was closed while
+/// attempting register a waiter.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Error {
+    Closed,
+    Busy,
+}
+
+pub const fn closed() -> Result<(), Error> {
+    Err(Error::Closed)
+}
+
+pub const fn registered() -> Result<(), Error> {
+    Ok(())
+}
+
+pub const fn busy() -> Result<(), Error> {
+    Err(Error::Busy)
+}
```

Comment:
do these need to be public?
```suggestion
const fn closed() -> Result<(), Error> {
    Err(Error::Closed)
}

const fn registered() -> Result<(), Error> {
    Ok(())
}

const fn busy() -> Result<(), Error> {
    Err(Error::Busy)
}
```